pkg/reconciler/members.go

```diff
@@ -225,19 +225,27 @@ func handleCreatingMembers(
 		)
 		return
 	}
+
 	// Perform setup on all of these members.
 	var wgSetup sync.WaitGroup
 	wgSetup.Add(len(creating))
 	for _, member := range creating {
 		go func(m *kdv1.MemberStatus) {
 			defer wgSetup.Done()
-			// If setup package is not present, skip doing any setup.
+
 			if setupUrl == "" {
-				// Set a temporary state used below so we won't send notifies
-				// to this member yet.
-				m.State = string(memberConfigured)
+				m.State = string(memberReady)
```

Comment:
The notification for roles with no config package will be done right away instead of waiting for another pass. That's the only reason.

Reply:
This particular notification is to tell already-ready node A that new node B has been added. We don't want to notify node B about node B being added, so that's why we do the two-phase change to its readiness state.

In practice I think this still ends up working out OK since the notification code itself will skip notifying node B as it doesn't have a setup package. But anyway it seems like we should have just left the two-phase change as it was.